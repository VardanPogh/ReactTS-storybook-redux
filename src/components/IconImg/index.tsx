import React from 'react'
import { makeStyles } from '@material-ui/core/styles'
import clsx from 'clsx'

const useStyles = makeStyles(() => ({
  main: {
    width: '100%',
  },
  title: {
    marginTop: 30,
    marginBottom: 0,
    color: ' #545454',
    fontSize: 24,
    textAlign: 'center',
  },
  subTitle: {
    margin: '16px 0',
    color: ' #545454',
    fontSize: 18,
    textAlign: 'center',
  },
  image: {
    maxWidth: 220,
    maxHeight: 220,
    margin: 'auto',
    display: 'block',
  },
}))

interface Props {
  image: any
  text: string
  title?: string
  className?: string
}

const IconImg: React.FC<Props> = ({ image, text, title, className }) => {
  const classes = useStyles()

  return (
    <div className={clsx(classes.main, className)}>
      <img src={image} alt="icon_image" className={classes.image} />
      {title && <p className={classes.title}>{title}</p>}
      <p className={classes.subTitle}>{text}</p>
    </div>
  )
}
export default IconImg
