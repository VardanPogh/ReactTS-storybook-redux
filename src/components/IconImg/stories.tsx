import React from 'react'
import { withKnobs, text, files } from '@storybook/addon-knobs'

import IconImg from '.'

export default {
  title: 'IconImg',
  component: IconImg,
  decorators: [withKnobs],
}

export const Default: React.FC = () => (
  <IconImg
    image={files('Icon/Image', '', [])}
    title={text('Title', 'Dr Jonh Smith')}
    text={text(
      'Description',
      'Lorem ipsum dolor, sit amet consectetur adipisicing elit. Error quae rerum maxime!'
    )}
  />
)
