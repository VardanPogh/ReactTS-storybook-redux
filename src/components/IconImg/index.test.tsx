import React from 'react'
import { render } from '@testing-library/react'
import IconImg from '.'

describe('IconImg Test', () => {
  test('Load IconImg', () => {
    const { container } = render(
      <IconImg
        image="https://encrypted-tbn0.gstatic.com/images?q=tbn%3AANd9GcTQEZrATmgHOi5ls0YCCQBTkocia_atSw0X-Q&usqp=CAU"
        title="Dr Jonh Smith"
        text="Lorem ipsum dolor, sit amet consectetur adipisicing elit. Error quae rerum maxime!"
      />
    )
    expect(container).toMatchSnapshot()
  })
})
