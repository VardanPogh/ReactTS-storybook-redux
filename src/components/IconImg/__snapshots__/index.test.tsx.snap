// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconImg Test Load IconImg 1`] = `
<div>
  <div
    class="makeStyles-main-1"
  >
    <img
      alt="icon_image"
      class="makeStyles-image-4"
      src="https://encrypted-tbn0.gstatic.com/images?q=tbn%3AANd9GcTQEZrATmgHOi5ls0YCCQBTkocia_atSw0X-Q&usqp=CAU"
    />
    <p
      class="makeStyles-title-2"
    >
      Dr Jonh Smith
    </p>
    <p
      class="makeStyles-subTitle-3"
    >
      Lorem ipsum dolor, sit amet consectetur adipisicing elit. Error quae rerum maxime!
    </p>
  </div>
</div>
`;
